server/models/Availability.js

```diff
@@ -0,0 +1,67 @@
+const mongoose = require("mongoose");
+
+const availableDaySchema = new mongoose.Schema({
+  sitterId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: "User"
+  },
+  available_day: {
+    type: String,
+    required: true,
+    enum: ["Mon", "Tue", "Wed", "Thu", "Fri"]
+  },
+  start_hour: {
+    type: Number,
+    required: true,
+    min: 1,
+    max: 24
+  },
+  end_hour: {
+    type: Number,
+    required: true,
+    validate: {
+      validator: function (endHour) {
+        return endHour >= this.start_hour && endHour <= 24;
+      }
+    }
+  },
+  start_min: {
+    type: Number,
+    required: true,
+    min: 0,
+    max: 59
+  },
+  end_min: {
+    type: Number,
+    required: true,
+    validate: {
+      validator: function (endMin) {
+        if (this.end_hour === this.start_hour) {
+          return endMin > this.start_min;
+        }
+        return endMin >= 0 && endMin < 60;
+      }
+    }
```

Comment:
Hi Uger, 
Thank you so much for the detailed guidance! 
For the use case,  I was wondering since the Calendarly photo of the ticket shows that the user will be asked to select a day of a week, the hour, and minutes directly, I initially designed to use string as day, and number as hours and minutes. 
If the start and end are Date type, I guess the user will be required to put all the year/month/day/hr/min information, correct? If that is the case, I plan to format the input with `new Date()` before trimming since it gives the same output regardless of the format of the input, but if the user put the whole year/month/day/hr/min in different order, the hour and minutes we extracted could be wrong, so my question is, do we have to require the user to put the start and end time along with all the year and month info in a specific format? Thank you! @ugur-akin

Reply:
Hi @sunmengyue,
> If the start and end are Date type, I guess the user will be required to put all the year/month/day/hr/min information, correct?

Not necessarily. You can always decouple the UI from the way your app works in the background - we can convert two numbers into a date, either in frontend or backend, like so: 
```
const dateFromNumbers = new Date(0, 0, 0, hoursAsNum, minsAsNum, 0);
```
So you can still let the users interact with your app the way you intended in the first place, using `Date`s would be a matter of convenience for your programming needs 👍🏼

Reply:
Thank you @ugur-akin ,

Just to double check for setter functions, I tried to trim the time and return only hour and minutes in the format of a string, but since the my start and end are Date type, the set does not pass the validation, nor did it pass when I change the type to be string, so I guess we can omit the setter function here and deal with the format somewhere else?

Reply:
Have you tried returning a `Date` from your setter instead? I'd encourage using `Date`s rather than strings for the reasons I detailed previously.

Also, what validation did it fail? Any comparison between `start/end` should work seamlessly if we do everything correctly. It could be a type mismatch if you tried returning a `string` when mongoose expected a `Date` 👍🏼